### ISSUE TITLE
run_cifar_distill.sh

### ISSUE THREAD

**Issue by JinYu1998** [zhengli97/CTKD#2]
Why is there no code for CTKD in the run_cifar_distill.sh file? And the code about 14 lines of DKD seems to be wrong!

**Comment by zhengli97** [comment]
The CTKD method has been integrated into other distillation methods. When you set `--have_mlp 1` and run the script `run_cifar_distill.sh`, it can automatically learn the temperature during training. You can obtain the vanilla distillation results without CTKD as long as you set `--have_mlp 0`.

**Comment by JinYu1998** [comment]
OK, thanks for your replay. I still have some questions:
1.  if I want to try vanilla CTKD, do I only need to set `--distill kd`?
2. The paper mentions using the Gradient Reversal Layer to learn a suitable temperature during training, so is it also applicable to other hyperparameters? For example, weights in loss(e.g. `KL Div`).

Thanks again for your reply!

**Comment by zhengli97** [comment]
1. Yes.
2. You can follow a similar idea by training a learnable module to generate appropriate hyperparameters for the network. But using GRL is probably not a good way to do it, because suitable hyperparameters may not be learned by inversely increasing the distillation loss. Anyway, you can try it.

**Comment by zhengli97** [comment]
Hi~ I will close this issue tomorrow if you have no more questions.

**Comment by JinYu1998** [comment]
> 

okk~~
thanks again for your reply. Have a great life~
